components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/common/InboundRunner.java

```diff
@@ -71,8 +75,13 @@ public void run() {
         while (!init) {
             log.debug("Waiting for the configuration context to be loaded to run Inbound Endpoint.");
             Boolean isSinglNode = ClusteringAgentUtil.isSingleNode();
+            Parameter clusteringPattern = ServiceReferenceHolder.getInstance().getConfigurationContextService().
+                    getServerConfigContext().getAxisConfiguration().getClusteringAgent().
+                    getParameter(CLUSTERING_PATTERN);
+            Boolean isWorkerManager = clusteringPattern != null && clusteringPattern.getValue() != null
+                    && clusteringPattern.getValue().toString().equals(CLUSTERING_PATTERN_WORKER_MANAGER);
             if (isSinglNode != null) {
-                if (!isSinglNode && !CarbonUtils.isWorkerNode() && !runOnManagerOverride) {
+                if (!isSinglNode && !CarbonUtils.isWorkerNode() && !runOnManagerOverride && isWorkerManager) {
```

Comment:
For this fix, it does not do any changes for the init flag (while loop is running based on that). It just check the node's type (ie. whether it is a worker/manger node), and if it is a manager node, set execute to false and print message. This fix is to  identify whether this a manager node or not considering the clustering pattern.